core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -184,14 +186,26 @@ func (c *OutgoingConnectorHandler) AwaitConnection(ctx context.Context) (string,
 				c.lggr.Warnw("failed to await connection to gateway node, retrying", "selectedGateway", gateway, "error", err)
 				continue
 			}
+
+			c.lggr.Debugw("connected successfully", "selectedGateway", gateway)
 			return gateway, nil
 		}
 	}
 }
 
+// allGatewaysAttempted checks if all available gateways have been attempted.
+func allGatewaysAttempted(attempts map[string]int) bool {
+	for _, count := range attempts {
+		if count == 0 {
+			return false
+		}
+	}
+	return true
+}
+
 // attemptGatewayConnection waits to connect to a gateway with a new child context
 func (c *OutgoingConnectorHandler) attemptGatewayConnection(ctx context.Context, gateway string) error {
-	timeout := (defaultFetchTimeoutMs / 4) * time.Millisecond
```

Comment:
Why is this changing? Was there a bug here?

Reply:
5 seconds was too long, was causing downstream timeouts.